sklearn/preprocessing/tests/test_data.py

```diff
@@ -2224,3 +2225,218 @@ def test_quantile_transform_valid_axis():
 
     assert_raises_regex(ValueError, "axis should be either equal to 0 or 1"
                         ". Got axis=2", quantile_transform, X.T, axis=2)
+
+
+def _generate_random_features_matrix(n_values=3, size=10):
+    rng = np.random.RandomState(0)
+    X = rng.randint(n_values, size=(size, n_values))
+    return X
+
+
+def test_unary_encoder():
+    X = np.arange(5).reshape(-1, 1)
+    enc = UnaryEncoder(5)
+    Xt = enc.fit_transform(X)
+    assert_array_equal(Xt, [[0, 0, 0, 0],   # 0
+                            [1, 0, 0, 0],   # 1
+                            [1, 1, 0, 0],   # 2
+                            [1, 1, 1, 0],   # 3
+                            [1, 1, 1, 1]])  # 4
+    Xt2 = enc.transform(X)
+    assert_array_equal(Xt2, Xt)
+
+
+def test_unary_encoder_stack():
+    # multiple input features stack to same output
+    n_values = np.random.randint(2, 10)
```

Comment:
We tend to prefer deterministic tests, using `rng = np.random.RandomState(0)` and `rng.randint` instead of `np.random.randint`.